web/src/activity.ts

```diff
@@ -21,6 +21,7 @@ const post_presence_response_schema = z.object({
             }),
         )
         .optional(),
+    presence_last_update_id: z.number().optional(),
```

Comment:
I think we don't need to tag this as optional, if we're passing the parameters that should make it always present.

Reply:
We need this optional, just like `presences` just above is `optional()`, because in the `ping_only: true` path, the server doesn't return these params.

And for that reason, in `send_presence_to_server()` below you have this block for handling the optionality in `ping_only: true` vs `false:
```
            if (redraw) {
                assert(
                    data.presences !== undefined,
                    "Presences should be present if not a ping only presence request",
                );
                assert(
                    data.server_timestamp !== undefined,
                    "Server timestamp should be present if not a ping only presence request",
                );
                assert(
                    data.presence_last_update_id !== undefined,
                    "Presence last update id should be present if not a ping only presence request",
                );

                presence.set_info(
                    data.presences,
                    data.server_timestamp,
                    data.presence_last_update_id,
                );
                redraw();
            }
```

Reply:
Ahh, yes, sounds good. This kind of detail can be good to record in a comment.

Reply:
Added a comment in an extra commit.